batcher/aligned-batcher/src/lib.rs

```diff
@@ -1238,4 +1241,41 @@ impl Batcher {
             .inspect_err(|e| warn!("Failed to get gas price: {e:?}"))
             .ok()
     }
+
+    pub(crate) async fn send_batch_inclusion_data_responses(
+        &self,
+        finalized_batch: Vec<BatchQueueEntry>,
+        batch_merkle_tree: &MerkleTree<VerificationCommitmentBatch>,
+    ) -> Result<(), BatcherError> {
+        for (vd_batch_idx, entry) in finalized_batch.iter().enumerate() {
+            let batch_inclusion_data = BatchInclusionData::new(vd_batch_idx, batch_merkle_tree);
+            let response = ResponseMessage::BatchInclusionData(batch_inclusion_data);
+
+            let serialized_response = cbor_serialize(&response)
+                .map_err(|e| BatcherError::SerializationError(e.to_string()))?;
+
+            let Some(ws_sink) = entry.messaging_sink.as_ref() else {
+                return Err(BatcherError::WsSinkEmpty);
+            };
+
+            let sending_result = ws_sink
+                .write()
+                .await
+                .send(Message::binary(serialized_response))
+                .await;
+
+            match sending_result {
+                Err(Error::ConnectionClosed) | Err(Error::Io(_)) => {
+                    self.metrics.broken_sockets_on_latest_batch.inc();
+                    error!("Error while sending the batch response, socket connection was closed!");
```

Comment:
I think this error message might be misleading, we are checking for other errors than the closed connection, maybe is worth to separate the cases or change the message to log the actual error.

Reply:
I agree! Just for the record, `IO` errs means a general error with the connection. This happens whenever you try to read or write from/to a socket that has been abnormally disconnected. The `ConnectionClosed` means that the underlying connection was gracefully closed. I'll add a comment about that in the code.

Reply:
Updated!